src/aggregate/aggregate_exec.c

```diff
@@ -790,6 +811,18 @@ int prepareExecutionPlan(AREQ *req, QueryError *status) {
     req->profileParseTime = rs_wall_clock_diff_ns(&req->initClock, &parseClock);
   }
 
+  // Currently supporting OOM policy only in standalone env
+  if (req->reqConfig.oomPolicy != OomPolicy_Ignore && !isClusterEnv())
+  {
+    // OOM guardrail
+    // Hold the GIL if necessary before check OOM
+    bool background = req->reqflags & QEXEC_F_RUN_IN_BACKGROUND;
+    if (estimateOOM(sctx->redisCtx, background)) {
+      RedisModule_Log(sctx->redisCtx, "notice", "Not enough memory available to execute the query");
+      QueryError_SetError(status, QUERY_EOOM, NULL);
```

Comment:
Use the code only API
```suggestion
      QueryError_SetCode(status, QUERY_EOOM);
```